test-with-jest/src/fakes/fake-in-mem-db.spec.ts

```diff
@@ -0,0 +1,61 @@
+// fake-in-mem-db.spec.ts
+class FakeDatabase {
+  private data: Record<string, any>;
+
+  constructor() {
+    this.data = {};
+  }
+
+  save(key: string, value: any): void {
+    this.data[key] = value;
+  }
+
+  get(key: string): any {
+    return this.data[key];
+  }
+}
+
+// Function to test
+function someTestFunction(db: FakeDatabase, key: string, value: any): any {
+  db.save(key, value);
+  return db.get(key);
+}
+
+describe('someTestFunction', () => {
+  let fakeDb: FakeDatabase;
+  let testKey: string;
+  let testValue: any;
+
+  beforeEach(() => {
+    fakeDb = new FakeDatabase();
+    testKey = 'testKey';
+    testValue = {
+      first: 'John',
+      last: 'Jones',
+      lastUpdated: new Date().toISOString(),
+    };
+
+  });
+
+  afterEach(() => {
+    // Clear all mocks
+    jest.clearAllMocks();
```

Comment:
I recommend `resetAllMocks` instead - otherwise you may run into issues when trying to mock something that was already mocked.

The standard pattern is:

setup: create your mocks / spies
test: verify state / output
teardown: `unmock` or remove all mocks

Reply:
ok